lib/ts/recipe/emailverification/components/features/emailVerification/index.tsx

```diff
@@ -110,7 +114,7 @@ export const EmailVerification: React.FC<Prop> = (props) => {
 
     const signOut = useCallback(async (): Promise<void> => {
         const session = Session.getInstanceOrThrow();
-        await session.signOut(props.userContext);
+        await session.signOut({ userContext: props.userContext ?? userContext });
```

Comment:
this type of check for `props.userContext` needs to be done in all places in this file. In fact, [here](https://github.com/supertokens/supertokens-auth-react/blob/b6b563b84ead4e4fb71d7718f33c53e06900568e/lib/ts/recipe/emailverification/components/features/emailVerification/index.tsx#L46) you can just assign userContext like 
```
let userContext = useUserContext()
if (props.userContext !== undefined) {
   userContext = props.userContext
}
```

Reply:
Make sure you do this everywhere where we use `useUserContext()` (can do a global search for this)

Reply:
This was the only place where `userContext` is also being passed as prop and `useUserContext` has been used.

Reply:
done